src/applications/personalization/profile/components/notification-settings/NotificationGroup.jsx

```diff
@@ -1,27 +1,77 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import { selectGroupById } from '@@profile/ducks/communicationPreferences';
 import { selectCommunicationPreferences } from '@@profile/reducers';
 
 import NotificationItem from './NotificationItem';
+import { BLOCKED_NOTIFICATION_IDS, NOTIFICATION_GROUPS } from '../../constants';
+import { useFeatureToggle } from '~/applications/personalization/hooks/useFeatureToggle';
 
-const NotificationGroup = ({ children, groupName, itemIds }) => {
-  return (
+const shouldRenderGroup = ({
+  groupId,
+  profileShowPaymentsNotificationSetting,
+  profileShowMhvNotificaionSettings,
+}) => {
+  if (groupId === NOTIFICATION_GROUPS.PAYMENTS) {
+    return profileShowPaymentsNotificationSetting;
+  }
+
+  if (groupId === NOTIFICATION_GROUPS.GENERAL) {
+    return profileShowMhvNotificaionSettings;
+  }
+
+  return true;
+};
```

Comment:
This is the guys of the conditional logic that tests to see if a group should render or not.

The whole payment and general group should not render if their respective toggle is turned OFF